naga/src/valid/expression.rs

```diff
@@ -224,7 +224,7 @@ impl super::Validator {
                 crate::TypeInner::Scalar { .. } => {}
                 _ => return Err(ConstExpressionError::InvalidSplatType(value)),
             },
-            _ if global_expr_kind.is_const(handle) || !self.allow_overrides => {
```

Comment:
Removing this will allow override-expressions through. `allow_overrides` didn't only mean "allow override variables" but also override-expressions.

I think we can rename the bool to `fully_evaluated_global_expressions`. If it's true, we check that all expressions in `global_expressions` have been fully evaluated (it would be a superset of the current `overrides_resolved`).

Reply:
I added this back, with the appropriate polarity change.